nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -0,0 +1,235 @@
+   * parameters given in the annotation must be one of the fields of the class or its super classes.
+   *
+   * @return Returns true, if the annotation conforms to the syntax rules.
+   */
+  protected boolean validateAnnotationSyntax(
+      Set<String> content,
+      NullAway analysis,
+      MethodTree tree,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol) {
+    if (content == null) {
+      return true;
+    } else {
+      String message;
+      if (content.isEmpty()) {
+        // we should not allow useless annotations.
+        message =
+            "empty @"
+                + annotName
+                + " is the default precondition for every method, please remove it.";
+        state.reportMatch(
+            analysis
+                .getErrorBuilder()
+                .createErrorDescription(
+                    new ErrorMessage(ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID, message),
+                    tree,
+                    buildDescriptionFromChecker(tree, analysis),
+                    state));
+        return false;
+      } else {
+        for (String fieldName : content) {
+          if (fieldName.contains(".")) {
+            if (!fieldName.startsWith(THIS_NOTATION)) {
+              message =
+                  "currently @"
+                      + annotName
+                      + " supports only class fields of the method receiver: "
+                      + fieldName
+                      + " is not supported";
+
+              state.reportMatch(
+                  analysis
+                      .getErrorBuilder()
+                      .createErrorDescription(
+                          new ErrorMessage(
+                              ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID, message),
+                          tree,
+                          buildDescriptionFromChecker(tree, analysis),
+                          state));
+              return false;
+            } else {
+              fieldName = fieldName.substring(fieldName.lastIndexOf(".") + 1);
+            }
+          }
+          Symbol.ClassSymbol classSymbol = ASTHelpers.enclosingClass(methodSymbol);
+          Element field = getFieldFromClass(classSymbol, fieldName);
+          if (field == null) {
+            message =
+                "cannot find field [" + fieldName + "] in class: " + classSymbol.getSimpleName();
+            state.reportMatch(
+                analysis
+                    .getErrorBuilder()
+                    .createErrorDescription(
+                        new ErrorMessage(
+                            ErrorMessage.MessageTypes.ANNOTATION_VALUE_INVALID, message),
+                        tree,
+                        buildDescriptionFromChecker(tree, analysis),
+                        state));
+            return false;
+          }
+          if (field.getModifiers().contains(Modifier.STATIC)) {
+            message =
+                "cannot accept static field: ["
```

Comment:
I don't see any reason why we couldn't support static fields, at least in a follow-up.  Our dataflow analysis already tracks their nullability.  CF may run into issues with static fields, since it tries to be sound for multi-threading (I think).

@nimakarimipour maybe we can remove this error and open an issue to support static fields in a follow-up?  I don't think we should add anything more to this PR since it's already quite large

Reply:
Ok Sure, I removed it. I will make the issue for that after it's landed.